src/install/install.zig

```diff
@@ -15439,6 +15439,9 @@ pub const PackageManager = struct {
         packages_len_before_install: usize,
         log_level: Options.LogLevel,
     ) OOM!void {
+        if (!this.options.do.save_lockfile) {
```

Comment:
We should check for this flag before calling this function. The bug exists in the long chain of or's and and's used to create `should_save_lockfile`. `this.options.do.save_lockfile` is already checked, so likely it's not checked early enough.

Reply:
Moved there: https://github.com/oven-sh/bun/pull/17481/commits/eb649a4025124418171411f83565c2627e14a51e